extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -694,4 +742,95 @@ export class AmpAnalytics extends AMP.BaseElement {
   }
 }
 
+/**
+ * Keeps track of the third-party vendors' cross-domain iframes. Only one
+ * static instance will be created, which will keep and manage ALL amp-analytics
+ * tags.
+ */
+class AmpAnalyticsFrames {
+  constructor() {
+    this.requestsFrameMap = map();
+  }
+
+  /**
+   * Create a cross-domain iframe
+   * @param {!HTMLDocument} parent  The document node of the parent page
+   * @param {!string} requestsFrameUrl  The URL, including data hash if
+   * applicable, of the cross-domain iframe
+   * @return {!Element}
+   */
+  create(parent, requestsFrameUrl) {
+    // Warning: the scriptSrc URL below is only temporary. Don't check
+    // in before getting resolution on that.
+    const frame = createElementWithAttributes(parent, 'iframe', {
+      sandbox: 'allow-scripts',
+      name: JSON.stringify({
+        'scriptSrc': '/examples/analytics-3p-remote-frame-helper.js',
+      }),
+    });
+    loadPromise(frame).then(() => {
+      this.setIsReady(requestsFrameUrl);
+    });
+    frame.src = requestsFrameUrl; // Set AFTER loadPromise to avoid race
```

Comment:
I don't need to do it after the promise is done, just after the promise is set up. In other words I *don't* want the sequence to be: 1. src is set, 2. frame is loaded, 3. ready event is fired, 4. we set up something to listen for the ready event, because then the ready event may not get received by anything. I may be being overzealous here, given JavaScript's single-threadedness, though.